netlify/edge-functions/og-image.tsx

```diff
@@ -0,0 +1,262 @@
+/* eslint-disable no-console */
+import React from 'https://esm.sh/react@18.2.0';
+import { ImageResponse } from 'https://deno.land/x/og_edge/mod.ts';
+/**
+ * The matching from github user to their full name
+ */
+const fullName: Record<string, string> = {
```

Comment:
Do we need an edge function? It seems that a regular function could do the job.

Is there a way to not duplicate this?

Reply:
I searched a bit, but seems that functions are not able to support JSX which would make it much harder to define the image. Could be done with JSX object (`{ type: 'div', props: { children: 'hello, world', style: { color: 'black' },},},`) but not very good DX

The reason why I used edge function at first is because the package I'm using: https://github.com/ascorbic/og-edge is made to run on Netlify Edge Functions. Since the author works at Netlify I assumed he knows which kind of function is the best (not by best technical decision)

Reply:
For the non duplication aspect of the `fullName` object, I'm not sure.

It's a bit the issue with those functions: they are all independent. But we could expose it in a URL and fetch it like this

```
import { fullName } from 'https://mui.com/public/contributors.json'
```

Reply:
> I searched a bit, but seems that functions are not able to support JSX which would make it much harder to define the image. Could be done with JSX object ({ type: 'div', props: { children: 'hello, world', style: { color: 'black' },},},) but not very good DX

Right, https://www.larocque.dev/projects/generative-open-graph-images/ says the same thing. We would need to add an intermediate step that converts JSX to JS. No preferences. 

> It's a bit the issue with those functions: they are all independent. But we could expose it in a URL and fetch it like this

This https://github.com/mui/material-ui/pull/41188#discussion_r1534864164 seems simple enough as a solution.